### DISCUSSION TITLE
Advice for using MVTLayer to viz millions of points with lots of categories

### DISCUSSION THREAD

**Discussion by cornhundred** [visgl/deck.gl#5605] [Q&A]
Hi, I have a dataset with millions of points that have hundreds of different categories. I am currently using tippecanoe to generate a clustered vector tileset of the points using the `--cluster-distance` option to merge points into clusters. This works great for visualizing the data, but I would also like to create a version where only points from the same category are clustered together. I would also like to have deck.gl show different categories of points in different colors (e.g. a user would select which categories they want to see and which color to represent them with). So I have the following questions:

* should I create multiple tilesets for each point category or one tileset with different sublayers for each point category (is one approach better for deck.gl)?
* what is the best way to have a variable number of point categories rendered on deck.gl? Should I add/remove MVTLayers to the layers array ``` deckgl.setProps({layers: layers}); ``` (see [simplified example](https://observablehq.com/@cornhundred/deck-gl-mvt-layer-nr?collection=@cornhundred/deck-gl) with points from two categories being rendered using two MVTLayers)? Should I be adding/removing layers using renderSubLayers?

**Comment by padawannn** [comment]
- Both approaches have their advantages and disadvantages. If you have multiple tilesets for each point category you will need more HTTPS calls if you want to view all categories of data at once but each tile will weigh less so this approach is better if, for example, you have a filter by categories in your interface.
- instead of adding or removing layers from the array try to use the `visible` property https://deck.gl/docs/api-reference/core/layer#visible

**Reply by cornhundred** [reply]
Thanks @padawannn I'll see which one works better for my case.
